### PR TITLE
feat/DAH-2339 account settings react

### PR DESCRIPTION
**Pull request by tallulahkay**
DAH-2339

### CODE REVIEW

File: app/javascript/pages/account/account-settings.tsx [hunk 1]
```diff
@@ -1,13 +1,353 @@
...
+                defaultValue={user ? user.lastName : null}
+                error={personalInfoErrors.lastName}
+                register={personalInfoRegister}
+                validation={{ required: true }}
+              />
+            </fieldset>
+            <div className="px-4 pb-4">
+              <DOBField
+                required
+                prepend="For example, November 23, 1975 is 11-23-1975"
```
Comment (tallulahkay) [inline app/javascript/pages/account/account-settings.tsx:313]: @chadbrokaw @cliu02 need a prepend on DOBField

File: app/javascript/pages/account/DOBFieldset.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      <DateField
+        fieldKey="birthDay"
+        defaultDOB={defaultDOB}
+        error={error}
+        register={register}
+        required={required}
+      />
+      <YearField
+        defaultDOB={defaultDOB}
+        error={error?.birthYear !== undefined}
```
Comment (chadbrokaw) [inline app/javascript/pages/account/DOBFieldset.tsx:128]: Why is the error handling different between Datefield and Yearfield?

File: app/javascript/pages/account/FormSubmitButton.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+interface FormSubmitButtonProps {
+  loading: boolean
+  label: string
+}
+
+const FormSubmitButton = ({ loading, label }: FormSubmitButtonProps) => {
+  return (
+    <div className="flex justify-center">
+      <Button loading={loading} type="submit">
```
Comment (chadbrokaw) [inline app/javascript/pages/account/FormSubmitButton.tsx:12]: Can you switch this to use the Seeds version of Button?

**Review by chadbrokaw — APPROVED** [review]
Awesome! I left a few questions as comments but otherwise I think this is a great starting point while we wait for more fleshed out designs

File: app/javascript/pages/account/FormSubmitButton.tsx (hunk 3 above)
Comment (tallulahkay) [reply app/javascript/pages/account/FormSubmitButton.tsx:12]: The Seeds version doesn't take a loading prop

File: app/javascript/pages/account/account-settings.tsx [hunk 4]
```diff
@@ -1,13 +1,231 @@
...
+
+const AccountSettings = ({ profile }: { profile: User }) => {
+  const [user, setUser] = useState(null)
+
+  useEffect(() => {
+    const dob = profile?.DOB
+    if (dob) {
+      const parts = dob.split("-")
+      const birth = { birthYear: parts[0], birthMonth: parts[1], birthDay: parts[2] }
+      profile.dob = birth
```
[the remark this answers is not in the document]
Comment (tallulahkay) [reply app/javascript/pages/account/account-settings.tsx:197]: We receive DOB from the api that's a string and we need an object with the date, month, year - I've updated the naming to make it clearer

File: app/javascript/pages/account/DOBFieldset.tsx (hunk 2 above)
Comment (tallulahkay) [reply app/javascript/pages/account/DOBFieldset.tsx:128]: There's actually no need to be - I made on component that should handle all 3 fields now

File: app/javascript/pages/account/account-settings.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (chadbrokaw) [reply app/javascript/pages/account/account-settings.tsx]: Got it, thank you!

File: app/javascript/pages/account/FormSubmitButton.tsx (hunk 3 above)
Comment (chadbrokaw) [reply app/javascript/pages/account/FormSubmitButton.tsx:12]: Ah got it, sorry I missed that!

**Pull request merged**